src/data-root.ts

```diff
@@ -37,26 +41,69 @@ export async function update(
 ): Promise<void> {
   const apiEndpoint = setup.endpoints.api
 
-  // Construct UCAN for the API call
-  const jwt = await ucan.build({
-    audience: await api.did(),
-    issuer: await did.ucan(),
-    potency: "APPEND",
-    proof,
+  // Cancel previous updates
+  if (fetchController) fetchController.abort()
+  fetchController = new AbortController()
 
-    // TODO: Waiting on API change.
-    //       Should be `username.fission.name/*`
-    resource: ucan.decode(proof).payload.rsc
-  })
+  // Ensure peer connection
+  await ipfs.reconnect()
 
   // Make API call
-  await Promise.all([
-    ipfs.reconnect(),
-    fetch(`${apiEndpoint}/user/data/${cid}`, {
-      method: 'PATCH',
-      headers: {
-        'authorization': `Bearer ${jwt}`
-      }
-    })
-  ])
+  await fetchWithRetry(`${apiEndpoint}/user/data/${cid}`, {
+    headers: async () => {
+      const jwt = await ucan.build({
+        audience: await api.did(),
+        issuer: await did.ucan(),
+        potency: "APPEND",
+        proof,
+
+        // TODO: Waiting on API change.
+        //       Should be `username.fission.name/*`
+        resource: ucan.decode(proof).payload.rsc
+      })
+
+      return { 'authorization': `Bearer ${jwt}` }
+    },
+    retries: 100,
+    retryDelay: 5000,
```

Comment:
why a delay? wouldn't we want to retry immediately?

Reply:
I thought a delay couldn't hurt, to give the server a bit of breathing room, since the requests are already lasting so long 🤷 A request could easily last 5 minutes, so waiting another 5 seconds doesn't change much I guess 😄

Reply:
ah kk :+1: